### PR TITLE
Validate all wheels with twine check after building

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** To avoid creating wheels that are incompatible with PyPI, we should validate all wheels post build to ensure they can be safely uploaded when ready.

**Description of the Change:** Adds a `twine check` command to validate each wheel after the CIBuildWheel step completes.

**Benefits:** Ensures the wheels are PyPI compatible.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/wheel_linux_aarch64.yml [hunk 1]
```diff
@@ -167,6 +167,11 @@ jobs:
 
         run: python3 -m cibuildwheel --output-dir wheelhouse
 
+      - name: Validate wheels
+        run: |
+          python -m pip install twine
+          python -m twine check ./wheelhouse/*.whl
```
Comment (vincentmr) [inline .github/workflows/wheel_linux_aarch64.yml:173]: I'm unsure whether it matters, but the rest of the file writes `python3 -m`. So we may want to change
 ```suggestion
          python3 -m pip install twine
          python3 -m twine check ./wheelhouse/*.whl
```
for consistency.

File: .github/workflows/wheel_linux_ppc64le.yml [hunk 2]
```diff
@@ -167,6 +167,11 @@ jobs:
 
         run: python3 -m cibuildwheel --output-dir wheelhouse
 
+      - name: Validate wheels
+        run: |
+          python -m pip install twine
+          python -m twine check ./wheelhouse/*.whl
```
Comment (vincentmr) [inline .github/workflows/wheel_linux_ppc64le.yml:173]: Same as above.
```suggestion
          python3 -m pip install twine
          python3 -m twine check ./wheelhouse/*.whl
```

File: .github/workflows/wheel_linux_x86_64.yml [hunk 3]
```diff
@@ -186,6 +186,11 @@ jobs:
 
         run: python3.8 -m cibuildwheel --output-dir wheelhouse
 
+      - name: Validate wheels
+        run: |
+          python -m pip install twine
+          python -m twine check ./wheelhouse/*.whl
```
Comment (vincentmr) [inline .github/workflows/wheel_linux_x86_64.yml:192]: Same as above.
```suggestion
          python3.8 -m pip install twine
          python3.8 -m twine check ./wheelhouse/*.whl
```

File: .github/workflows/wheel_linux_aarch64.yml (hunk 1 above)
Comment (mlxd) [reply .github/workflows/wheel_linux_aarch64.yml:173]: Good point --- this may be platform specific. I'll commit all changes. Thanks.

**Review by vincentmr — APPROVED** [review]
Looks good to me. Cheers!

**Comment by mlxd** [comment]
[sc-35729]

**Pull request merged**